node/src/blockchain/blockchain_interface/test_utils.rs

```diff
@@ -1,70 +1,112 @@
-            .borrow_mut()
-            .remove(0)
+    fn get_transaction_fee_balance(
+        &self,
+        _address: Address,
+    ) -> Box<dyn Future<Item = U256, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
     }
 
-    fn get_service_fee_balance(&self, address: &Wallet) -> ResultForBalance {
-        self.get_masq_balance_params
-            .lock()
-            .unwrap()
-            .push(address.clone());
-        self.get_masq_balance_results.borrow_mut().remove(0)
+    fn get_service_fee_balance(
+        &self,
+        _address: Address,
+    ) -> Box<dyn Future<Item = U256, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
     }
 
-    fn get_block_number(&self) -> LatestBlockNumber {
-        self.get_block_number_results.borrow_mut().remove(0)
+    fn get_gas_price(&self) -> Box<dyn Future<Item = U256, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
     }
 
-    fn get_transaction_id(&self, address: &Wallet) -> ResultForNonce {
-        self.get_transaction_id_params
-            .lock()
-            .unwrap()
-            .push(address.clone());
-        self.get_transaction_id_results.borrow_mut().remove(0)
+    fn get_block_number(&self) -> Box<dyn Future<Item = U64, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
+    }
+
+    fn get_transaction_id(
+        &self,
+        _address: Address,
+    ) -> Box<dyn Future<Item = U256, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
+    }
+
+    fn get_transaction_receipt(
+        &self,
+        _hash: H256,
+    ) -> Box<dyn Future<Item = Option<TransactionReceipt>, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
+    }
+
+    fn get_transaction_receipt_batch(
+        &self,
+        _hash_vec: Vec<H256>,
+    ) -> Box<dyn Future<Item = Vec<TransactionReceiptResult>, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
+    }
+
+    fn get_contract(&self) -> Contract<Http> {
+        unimplemented!("not needed so far")
+    }
+
+    fn get_transaction_logs(
+        &self,
+        _filter: Filter,
+    ) -> Box<dyn Future<Item = Vec<Log>, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
+    }
+
+    fn submit_payables_in_batch(
+        &self,
+        _logger: Logger,
+        _chain: Chain,
+        _consuming_wallet: Wallet,
+        _fingerprints_recipient: Recipient<PendingPayableFingerprintSeeds>,
+        _affordable_accounts: Vec<PayableAccount>,
+    ) -> Box<dyn Future<Item = Vec<ProcessedPayableFallible>, Error = PayableTransactionError>>
+    {
+        unimplemented!("not needed so far")
     }
 }
 
 impl LowBlockchainIntMock {
-    pub fn get_transaction_fee_balance_params(mut self, params: &Arc<Mutex<Vec<Wallet>>>) -> Self {
+    pub fn get_transaction_fee_balance_params(mut self, params: &Arc<Mutex<Vec<Address>>>) -> Self {
```

Comment:
I used to like Utkarsh's idea about stopping to use the Wallet for actually needing it just to refer to the address of a wallet. The concern of unnecessary bulkiness and the related tool-set variety has been sound, the big problem I see these days  (and I repeat it to myself with bigger and bigger stress with every new place where I see this idea implemented) is the incompetence of this `Address` type for the replacement. I actually hate seeing this vague term as it can potentially represent dozens of different kinds of addresses. The problem is this is actually a type alias we take over from the web3 library (seems also wrong, as we won't want to be dependent on them forever). 

Anyway, you might say I'm a pain in the neck (what I sometime actually can be haha), but I'd like to see our own type alias like `type WalletAddress` = H180;` or even a very primitive struct wrapping it around, which would also allow us implement some methods for it later on, if need. 
I feel like it'd be a mess trying to do it in this card. But can I ask you to maybe write one. When it comes to whether I agree with these changes in the PR for the meantime, I'd say I'd prefer you not to do that but honestly, I'm quite indifferent at the same time. 

Alert, Dan's opinion would be worth bothering him.